mantidproject/mantid
# Upgrade Qt to 5.15.8

**Description of work.**
Upgrade Qt version from 5.15.4 to 5.15.8. It was previously pinned to 5.15.4 to avoid GUI rendering problems, but these seem to have been resolved now. This is required before #34820 can be addressed.
Moving to 5.15.8 has meant that #35029 has become a problem again.

Other changes:
- Old Poco pin was holding back Qt (I think), so this has been updated. It has been restricted to avoid v1.12.2 to 1.12.4 because they added an unguarded `#define NOMINMAX` that introduced a compiler warning because we define it ourselves in WindowsSetup.cmake. They've added a guard around it on their devel branch, so it should be fine in later versions.
- Remove pin for OpenSSL. This was also required to resolve the conda environment.
- Another compiler warning cropped up on Windows to do with `strstream` depreciation, which should be fixed now.

**To test:**
This pipeline build passed all stages with no issues:
https://builds.mantidproject.org/job/build_packages_from_branch/335/
Install and test the packages from this build, see #34964 and #35029 for details about how the GUI was broken when Qt moved to 5.15.6. Check that those two issues are OK and try to break it in other ways.

Fixes #35194 
Fixes #35268 
Fixes #35029 

*This does not require release notes* because **it's a dependency update that should be invisible to the user.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
